CARET_trace/src/ros_trace_points.cpp

```diff
@@ -51,6 +52,81 @@
 std::unique_ptr<std::thread> trace_node_thread;
 thread_local bool trace_filter_is_rcl_publish_recorded;
 
+using std::string;
+using std::vector;
+static bool ignore_rcl_timer_init = false;
+
+static vector<string> string_split(string & str, char delim)
+{
+  using std::stringstream;
+  vector<string> elems;
+  stringstream ss(str);
+  string item;
+  while (getline(ss, item, delim)) {
+    if (!item.empty()) {
+      elems.push_back(item);
+    }
+  }
+  return elems;
+}
+
+static bool is_ros2_launch_command()
+{
+  // Caret_trace's <TraceNode> should not record rcl_timer_init() for command line
+  // "python3" launch.
+  // This is because there are multiple processes that repeat spin_once() in a short time
+  // in the process started by "python3".
+  // If this is not excluded, the end time will increase as time passes
+  // after Caret_trace starts.
+  using std::ifstream;
+  using std::ios;
+  using std::istreambuf_iterator;
+  using std::to_string;
+  pid_t pid = getpid();
+  string fileName = "/proc/" + to_string(pid) + "/cmdline";
+
+  ifstream ifs(fileName, ios::in | ios::binary);
+  if (!ifs) {
+    return false;
+  }
+  istreambuf_iterator<char> it_ifs_begin(ifs);
+  istreambuf_iterator<char> it_ifs_end{};
+  vector<char> input_data(it_ifs_begin, it_ifs_end);
+  if (!ifs) {
+    ifs.close();
+    return false;
+  }
+  ifs.close();
+
+  vector<string> cmd_line;
+  auto itr_begin = input_data.begin();
+  auto itr_find = input_data.begin();
+  for (;;) {
+    itr_find = find(itr_begin, input_data.end(), 0x00);
+    if (itr_find == input_data.end()) {
+      break;
+    }
+    string output_string(itr_begin, itr_find);
+    cmd_line.push_back(output_string);
+    itr_begin = itr_find + 1;
+  }
+
+  // Check if the right side of the last '/' in cmdline[0] matches 'python3'.
+  if (cmd_line.size() < 2) {
```

Comment:
You can change this condition becose following process requires only one element.
```suggestion
  if (cmd_line.size() < 1) {
```

Reply:
It has changed